test/tuple.jl

```diff
@@ -807,3 +807,14 @@ namedtup = (;a=1, b=2, c=3)
 @test Val{Tuple{Vararg{T,4}} where T} === Val{Tuple{T,T,T,T} where T}
 @test Val{Tuple{Int64, Vararg{Int32,N}} where N} === Val{Tuple{Int64, Vararg{Int32}}}
 @test Val{Tuple{Int32, Vararg{Int64}}} === Val{Tuple{Int32, Vararg{Int64,N}} where N}
+
+@testset "circshift" begin
+    t1 = (1, 2, 3, 4, 5)
+    t2 = (1, 'a', -7.0, 3)
+    @test Base.circshift(t1, 2) == (4, 5, 1, 2, 3)
+    @test Base.circshift(t1, Val(2)) == (4, 5, 1, 2, 3)
+    @test Base.circshift(t2, 3) == ('a', -7.0, 3, 1)
+    @test Base.circshift(t2, Val(3)) == ('a', -7.0, 3, 1)
```

Comment:
Add `@inferred` to some of the tests?

Reply:
Thank you, @stevengj. This is a good test. I've now included tests for type inference. The only case where type cannot be inferred is when we have mixed tuples (i.e., when not all elements have the same type) and the shift amount is not known at compile time.